File: src/me/riseremi/cards/Effect.java
package me.riseremi.cards;

import lombok.Getter;
import lombok.Setter;

/**
 *
 * @author Remi Weiss <riseremi at icloud.com>
 */
public class Effect {

    @Getter @Setter private BasicCard.EffectType effectType;
    @Getter @Setter private Object value;

    public Effect() {
        this.effectType = BasicCard.EffectType.NONE;
        this.value = 0;
    }

    @Override
    public String toString() {
        return effectType.name() + " = " + value.toString();
    }

    
}

File: src/me/riseremi/cards/Effect.kt
package me.riseremi.cards

/**
 *
 * @author riseremi <riseremi at icloud.com>
</riseremi> */
class Effect(private val effectType: Card.Companion.EffectType, private val value: Int) {

    override fun toString(): String = effectType.name + " = " + value.toString()

    // NOTE: for compatibility only
    // TODO: remove in flavor of new Card class
    fun getEffectType(): Card.Companion.EffectType = effectType

    // NOTE: for compatibility only
    // TODO: remove in flavor of new Card class
    fun getValue(): Int = value
}
